src/query/execute_schema.c

```diff
@@ -10016,6 +10021,203 @@ do_alter_change_tbl_comment (PARSER_CONTEXT * const parser, PT_NODE * const alte
   return error;
 }
 
+/*
+ * do_alter_change_col_comment() - change the column comment
+ *   return: Error code
+ *   parser(in): Parser context
+ *   alter(in/out): Parse tree of a PT_CHANGE_COLUMN_COMMENT clause
+ */
+static int
+do_alter_change_col_comment (PARSER_CONTEXT * const parser, PT_NODE * const alter_node)
+{
+  int error = NO_ERROR;
+  SM_ATTR_CHG_SOL change_mode = SM_ATTR_CHG_ONLY_SCHEMA;
+  SM_ATTR_PROP_CHG attr_chg_prop;
+  SM_ATTRIBUTE *found_attr = NULL;
+  SM_NAME_SPACE name_space = ID_NULL;
+  const PT_ALTER_CODE alter_code = alter_node->info.alter.code;
+  const char *entity_name = NULL;
+  PT_NODE *attr_node = NULL;
+  const char *attr_name = NULL;
+  PT_NODE *comment_node = NULL;
+  PARSER_VARCHAR *comment_str = NULL;
+  DB_OBJECT *class_obj = NULL;
+  DB_CTMPL *ctemplate = NULL;
+  MOP class_mop = NULL;
+  OID class_oid;
+  bool tran_saved = false;
+
+  assert (alter_code == PT_CHANGE_COLUMN_COMMENT);
+
+  OID_SET_NULL (&class_oid);
+  reset_att_property_structure (&attr_chg_prop);
+
+  entity_name = alter_node->info.alter.entity_name->info.name.original;
+  if (entity_name == NULL)
+    {
+      error = ER_UNEXPECTED;
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 1, "Expecting a class or virtual class name.");
+      goto exit;
+    }
+
+  class_obj = db_find_class (entity_name);
+  if (class_obj == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      error = er_errid ();
+      goto exit;
+    }
+
+  error = locator_flush_class (class_obj);
+  if (error != NO_ERROR)
+    {
+      /* don't overwrite error */
+      goto exit;
+    }
+
+  /* force exclusive lock on class, even though it should have been already acquired */
+  if (locator_fetch_class (class_obj, DB_FETCH_WRITE) == NULL)
+    {
+      error = ER_FAILED;
+      goto exit;
+    }
+
+  error = tran_system_savepoint (UNIQUE_SAVEPOINT_CHANGE_COLUMN_COMMENT);
+  if (error != NO_ERROR)
+    {
+      goto exit;
+    }
+  tran_saved = true;
+
+  attr_node = alter_node->info.alter.alter_clause.attr_mthd.attr_def_list;
+  while (attr_node != NULL)
+    {
+      ctemplate = dbt_edit_class (class_obj);
+      if (ctemplate == NULL)
+	{
+	  /* when dbt_edit_class fails (e.g. because the server unilaterally aborts us), we must record the associated
+	   * error message into the parser. Otherwise, we may get a confusing error msg of the form: "so_and_so is not a
+	   * class". */
+	  pt_record_error (parser, parser->statement_number - 1, alter_node->line_number, alter_node->column_number,
+			   er_msg (), NULL);
+	  error = er_errid ();
+	  goto exit;
+	}
+
+      attr_name = get_attr_name (attr_node);
+
+      comment_node = attr_node->info.attr_def.comment;
+
+      assert (comment_node != NULL && comment_node->node_type == PT_VALUE);
```

Comment:
Isn't it supposed to handle errors, not assert? There are also several assert below.

Reply:
Thank you. If there is a problem in this part, the do_alter_change_col_comment function will cause a problem to work, so I will handle the error.

Reply:
I will not add error handling as it is checked at the parsing step.

Reply:
Maybe the error is already at grammar paring step.
So, I think that assert is not bad.